narwhal/source_code_sharing/update_liferay_dxp_source_code_of_hotfixes.sh

```diff
@@ -25,24 +30,46 @@ function check_if_tag_exists {
 	fi
 }
 
-function check_ignore_zip_file {
-	local hotfix_zip_file="${1}"
-	local release_version="${2}"
+function check_ignore_via_argument {
+	local ignore_zip_file="${1}"
 
-	local file_url="${zip_directory_url}/${hotfix_zip_file}"
+	if [[ "x${ignore_zip_file}" =~ x*${hotfix_zip_file}* ]]
+	then
+		lc_log DEBUG "Ignoring '${file_url}'."
+
+		return "${LIFERAY_COMMON_EXIT_CODE_OK}"
+	else
+		lc_log DEBUG "The file is not ignored via command line argument."
+
+		return "${LIFERAY_COMMON_EXIT_CODE_BAD}"
+	fi
+}
+
+function check_ignore_via_file {
+	local ignore_zip_file="${1}"
 
-	if [[ "x${IGNORE_ZIP_FILES}" =~ x*${hotfix_zip_file}* ]]
+	if (grep -q "${hotfix_zip_file}" "${ignore_zip_file}")
 	then
 		lc_log DEBUG "Ignoring '${file_url}'."
 
 		return "${LIFERAY_COMMON_EXIT_CODE_OK}"
 	else
-		lc_log DEBUG "The file on '${file_url}' is not on the ignore list."
+		lc_log DEBUG "The file on '${file_url}' is not on the ${ignore_zip_file}."
 
 		return "${LIFERAY_COMMON_EXIT_CODE_BAD}"
 	fi
 }
 
+function check_ignore_zip_file {
+	local hotfix_zip_file="${1}"
+
+	local file_url="${zip_directory_url}/${hotfix_zip_file}"
+
+	check_ignore_via_argument "${IGNORE_ZIP_FILES}"
```

Comment:
Either there should be a return if the ignore statement is fulfilled or we should not use an upstream function for it.
Correct me if I am wrong and the downstream functions 'return' command stops processing the upstream function too.
I am referring to the upstream function as `check_ignore_zip_file` and downstream as `check_ignore_via_...`.